pkg/dao/resource.go

```diff
@@ -115,3 +116,10 @@ func (d *sqlResourceDao) All(ctx context.Context) (api.ResourceList, error) {
 	}
 	return resources, nil
 }
+
+func (d *sqlResourceDao) FirstByConsumerName(ctx context.Context, consumerName string) (api.Resource, error) {
```

Comment:
Why first?

Reply:
I just chose one(first) item to determine whether the resource exists for the consumer. Of course, we can choose to use the current [FindByConsumerName](https://github.com/openshift-online/maestro/blob/main/pkg/dao/resource.go#L20) to do that. The latter may need more memory footprint, call it will load all the resources on the consumer to the memory. What do you think of it?

Reply:
yeah, I think we need to document this with the reason. Also what if the resource is marked as deleted or deleting? do we still block?

Reply:
Yes, I added comments for the `FirstByConsumerName`.

If a resource is marked as deleted, we will still block the deletion.